sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -114,17 +116,18 @@ object FileFormatWriter extends Logging {
 
     val caseInsensitiveOptions = CaseInsensitiveMap(options)
 
-    val dataSchema = dataColumns.toStructType
-    DataSourceUtils.verifySchema(fileFormat, dataSchema)
-    DataSourceUtils.checkFieldNames(fileFormat, dataSchema)
+    val tableSchema = finalOutputSpec.tableSchema
```

Comment:
From line 112:
```
val dataColumns = finalOutputSpec.outputColumns.filterNot(partitionSet.contains)
```
We should filter out the partition columns from `tableSchema` as well

Reply:
I mean, the schema here should not include the partition columns. Otherwise it is breaking change and there can be bugs.